CenterForOpenScience/osf.io
# [ENG-361] Remove dead prereg code

## Purpose

We have completed the prereg challenge! With new reviewer changes being handled on v2 we don't need the old prereg code, so time to delete the old v1 code.

# Changes

- Remove special casing for permissions on prereg for views
- Remove special casing for checking out a prereg file
- Remove special casing for checking out a prereg file permission
- Remove admin `pre_reg` app
- Remove prereg challenge JSON schema
- Remove prereg landing page waffling
- Remove prereg analytics from admin dashboard
- Remove Prereg Challenge specific campaign code (while keeping OSF Preregistration)

## QA Notes

Check out admin app and draft registration process to ensure there's no new bugs. I am most worried about difficult/usually draft registrations using prereg features.

## Documentation

Not user facing

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-361
